handler.js

```diff
@@ -397,7 +397,7 @@ function discover(handler, req, head, body, cb) {
 
     var svcchan = null;
     var myHost = handler.channel.hostPort;
-    if (exitHosts.indexOf(myHost) === -1) {
+    if (exitHosts.indexOf(myHost) < 0 && req.headers.cn !== 'hyperbahn') {
```

Comment:
As I understand it correctly the discover call will be executed locally when it has been called by hyperbahn ie. forwarded. What would the response be if it has been forwarded to the wrong machine because of ring rebalance that has not completely propagated? Will it just work, will you get an empty list back, or is there an error?

As I described in the chat about how ringpop solves it you might want to double check you are an exithost even if the call has been forwarded and throw an error if you are not an exitnode.